src/python_testing/TC_ACL_2_11.py

```diff
@@ -113,24 +123,47 @@ async def test_TC_ACL_2_11(self):
+                    # if ID1 is null, it means it is a wildcard.  We need to read all attributes on the cluster
+                    if ID1 is NullValue:
+                        for attr_id, attribute in ALL_ATTRIBUTES[C1].items():
+                            if attr_id not in self.global_attributes:
+                                await self.read_single_attribute_expect_error(cluster=cluster, attribute=attribute, error=Status.AccessRestricted, endpoint=E1)
+                    else:
+                        attribute = ALL_ATTRIBUTES[C1][ID1]
+                        await self.read_single_attribute_expect_error(cluster=cluster, attribute=attribute, error=Status.AccessRestricted, endpoint=E1)
                 elif restriction_type == AccessControl.Enums.AccessRestrictionTypeEnum.kAttributeWriteForbidden:
-                    status = await self.write_single_attribute(attribute_value=attribute, endpoint_id=E1)
-                    asserts.assert_equal(status, Status.AccessRestricted,
-                                         f"Failed to verify ACCESS_RESTRICTED when writing to Attribute {ID1} Cluster {C1} Endpoint {E1}")
+                    if ID1 is NullValue:
+                        for attr_id, attribute in ALL_ATTRIBUTES[C1].items():
+                            if attr_id not in self.global_attributes:
+                                status = await self.write_single_attribute(attribute_value=attribute(), endpoint_id=E1, expect_success=False)
+                                if status is not Status.UnsupportedWrite:
+                                    asserts.assert_equal(status, Status.AccessRestricted,
+                                                         f"Failed to verify ACCESS_RESTRICTED when writing to Attribute {attr_id} Cluster {C1} Endpoint {E1}")
+                    else:
+                        attribute = ALL_ATTRIBUTES[C1][ID1]
+                        status = await self.write_single_attribute(attribute_value=attribute(), endpoint_id=E1, expect_success=False)
+                        asserts.assert_equal(status, Status.AccessRestricted,
+                                             f"Failed to verify ACCESS_RESTRICTED when writing to Attribute {ID1} Cluster {C1} Endpoint {E1}")
                 elif restriction_type == AccessControl.Enums.AccessRestrictionTypeEnum.kCommandForbidden:
-                    result = await self.send_single_cmd(cmd=command, endpoint=E1)
-                    asserts.assert_equal(result.status, Status.AccessRestricted,
-                                         f"Failed to verify ACCESS_RESTRICTED when sending command {ID1} to Cluster {C1} Endpoint {E1}")
+                    if ID1 is NullValue:
+                        for cmd_id, command in ALL_ACCEPTED_COMMANDS[C1].items():
+                            try:
+                                await self.send_single_cmd(cmd=command(), endpoint=E1, timedRequestTimeoutMs=1000)
```

Comment:
Does this actually work right?  Don't commands need payloads?  Will this auto-generate some default payload for you?

Reply:
It does work, thankfully.  Otherwise I'm not sure what I would have had to do to generically invoke each command.  It gets far enough for the access check to return AccessRestricted (what we are looking for) or another error related to the bogus payload.

Reply:
Yes, fortunately.  If I had to craft legit payloads for attribute writes and command invocations for each type of attribute/command, I don't know how I would pull that off.  I was relieved to find that the access checks happen before payload decoding (which makes sense).  This method does detect restricted commands.